packages/site/src/components/TokensView.js

```diff
@@ -0,0 +1,177 @@
+import { Route, Switch, useRouteMatch } from "react-router-dom";
+
+import React, { useState, useEffect } from "react";
+import { TokenForm } from "./TokenForm";
+import { InitialForm } from "./InitialForm";
+import {
+	getCoreColorsFromTokens,
+	getUpdatedTokensWithCoreColors,
+} from "../utils/coreColorTokens";
+
+export const TokensView = () => {
+	let match = useRouteMatch();
+
+	const [selectedBrand, setSelectedBrand] = useState({
+		action: "",
+		brand: "",
+	});
+
+	const supportsshowSaveFilePicker = () =>
+		undefined !== window.showSaveFilePicker;
+
+	const [tokens, setTokens] = useState({});
+	const [copied, setCopied] = useState(false);
+	const [copyText, setCopyText] = useState("");
+	const [canSaveFile] = useState(supportsshowSaveFilePicker);
+	const [coreColorTokens, setCoreColorTokens] = useState({});
+
+	// Handle the updates that are side effects of the
+	// check for browser support
+	useEffect(() => {
+		const beforeText = canSaveFile
+			? "Save JSON to File"
+			: "Copy JSON to Clipboard";
+		const afterText = canSaveFile ? "Saved!" : "Copied!";
+
+		if (copied) {
+			setCopyText(afterText);
+		} else {
+			setCopyText(beforeText);
+		}
+	}, [copied, copyText, canSaveFile]);
+
+	// Handle updating the tokens state when we update
+	// the core colors state to "link" the values.
+	useEffect(() => {
+		// const newTokens = getUpdatedTokensWithCoreColors( tokens, coreColorTokens );
+		// setTokens( newTokens );
+	}, [tokens, coreColorTokens]);
+
+	const handleUpdateBrand = (brand, action) => {
+		setSelectedBrand({
+			brand,
+			action,
+		});
+	};
+
+	const updateTokenValue = (tokenData, newValue) => {
+		const updateValueHelper = (tokensObj, updateStateFn) => {
+			tokensObj[tokenData.name].value = newValue;
+
+			updateStateFn(tokensObj);
+		};
+
+		if (tokenData.category !== "core-color") {
+			updateValueHelper(tokens, setTokens);
+		} else {
+			updateValueHelper(coreColorTokens, setCoreColorTokens);
+		}
+	};
+
+	const updateTokensWithCoreColors = () => {
+		const newTokens = getUpdatedTokensWithCoreColors(
+			tokens,
+			coreColorTokens
+		);
+		setTokens(newTokens);
+	};
+
+	const updateCoreColors = () => {
+		const newCoreColors = getCoreColorsFromTokens( tokens );
+
+		setCoreColorTokens(newCoreColors);
+	};
+
+	const fetchAndSetTokens = async (e) => {
+		const brand =
+			"create" === selectedBrand.action ? "default" : selectedBrand.brand;
```

Comment:
I would recommend converting all of these strings to Constants defined elsewhere (or enums if you integrate typescript). It's best practice so you only have to change a string in one spot if it every changes. Especially if you're going to use them to determine state